doc/tools/docgram/fullGrammar

```diff
@@ -2019,6 +2031,19 @@ tactic_atom: [
 | integer
 | reference
 | "()"
+| Prim.integer
+| Prim.string
+| Prim.qualid
+| "@" Prim.ident
+| "&" lident
+| "'" Constr.constr
+| "constr" ":" "(" Constr.lconstr ")"
+| "open_constr" ":" "(" Constr.lconstr ")"
+| "ident" ":" "(" lident ")"
+| "pattern" ":" "(" Constr.lconstr_pattern ")"
+| "reference" ":" "(" globref ")"
+| "ltac1" ":" "(" ltac1_expr_in_env ")"
+| "ltac1val" ":" "(" ltac1_expr_in_env ")"
```

Comment:
It seems likely that these extensions of `tactic_atom` should have been added to some other non-terminal instead...

Reply:
>  Perhaps it's better to clean up the syntax for other chapters first, in particular doing as much of the tactics chapter as we can for 8.12. And doing much less rewriting.

I insist that there is no rush in fixing the syntax in the tactics chapter. You keep coming back to the idea of fixing the syntax with very little rewriting but experience has shown again and again that updating the syntax without doing the associated rewriting is not possible or would create major regressions in the documentation.

Let me be clear: I'd rather focus on other improvements to the refman for 8.12 and won't review any significant PR about the tactics chapter in the coming weeks. If we don't finish replacing the `productionlist` for 8.12, fine, but let's then focus on coordinating an effort to improve the pages that are new in 8.12.

Reply:
If you want to defer reviewing further syntax updates until after 8.12 ships, that's fine.  I think we need more discussion before I can do much on Ltac2.  We can defer that discussion, too.  I think I could make a lot of progress on the tactics chapter in the meantime.  If you want my help for other doc improvements for 8.12, let me know what you have in mind.

Reply:
It will be more a matter of helping the volunteers if we can find some.